docs/setup/kubeedge_install_source.md

```diff
@@ -0,0 +1,165 @@
+# Setup from Source Code
+
+This guide provide steps which can be utilised to install KubeEdge Cloud and Edge side. At this point, we assume that you would have installed the [Pre-Requisite](kubeedge_install.md#prerequisite) for Cloud and Edge.
+
+## Setup Cloud Side (KubeEdge Master Node)
+
+1. Clone KubeEdge
+
+    + Setup [$GOPATH ](https://github.com/golang/go/wiki/SettingGOPATH) to clone the KubeEdge repository in the `$GOPATH`.
+
+        ```shell
+        git clone https://github.com/kubeedge/kubeedge.git $GOPATH/src/github.com/kubeedge/kubeedge
+        cd $GOPATH/src/github.com/kubeedge/kubeedge
+        ```
+
+2. Generate Certificates
+
+   RootCA certificate and a cert/ key pair is required to have a setup for KubeEdge. Same cert/ key pair can be used in both cloud and edge.
+
+    ```bash
+    $GOPATH/src/github.com/kubeedge/kubeedge/build/tools/certgen.sh genCertAndKey edge
+    ```
+
+    The cert/ key will be generated in the `/etc/kubeedge/ca` and `/etc/kubeedge/certs` respectively, so this command should be run with root or users who have access to those directories. Copy these files to the corresponding edge side server directory.
+
+3. Compile Cloudcore
+
+
+   + Firstly, make sure `gcc` is already installed on your host. You can verify it via:
```

Comment:
Make sure a C compiler is installed on your host. The installation is tested with `gcc` and `clang`.